Userland/Demos/Mouse/main.cpp

```diff
@@ -148,14 +151,25 @@ class MainFrame final : public GUI::Frame {
         start_timer(500);
     }
 
+    void doubleclick_event(GUI::MouseEvent& event) override
+    {
+        m_double_click_buttons |= event.button();
+        update();
+        if (has_timer())
+            stop_timer();
+        start_timer(500);
+    }
+
 private:
     unsigned m_buttons;
     unsigned m_wheel_delta_acc;
     bool m_show_scroll_wheel;
+    unsigned m_double_click_buttons;
```

Comment:
Looking at other instances in the code base (/unsigned m_/), I think this should be zero-initialized.

Reply:
Isn't the `m_double_click_buttons { 0 }` below initializing to 0, in the same way as `unsigned m_buttons`?

Reply:
I guess looking at it, the possibility is that there would be another constructor added that doesn't zero-initialize it? Maybe it's a good practice thing.

Maybe I'm wrong. Thanks for the other changes you made.

Reply:
That makes sense. I'm a total beginner in C++, so I'm open for any change :D But it also makes sense to follow what is done for the other members in this class, right?

Reply:
I found this https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#other-punctuation. According to this doc it would seem best to zero-initialize the member variable instead of doing it in the constructor.

Reply:
I added a new commit to change the initialization,  https://github.com/SerenityOS/serenity/pull/16219/commits/7554a86631434098b98899b3cf2b44658d7b6ff9. I hope that's appropriate.

Reply:
Yes good idea! However, I'd point you to https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md for the "Iterate excessively on your design across multiple commits" rule. :D

The issue is that in one commit (Draw blue color for longer to indicate double click) you create m_double_click_buttons and initialize it the 'old' way, and then in another commit in the same PR you correct this.

You should initialize it in the correct manner in the original commit, and then you could have this latter commit where you apply the fix to the other 3 variables.

Reply:
You are right; I have read that guideline, but didn't think I was violating it, since the change need to be applied to other members as well. I fixed it now by _first_ adding a commit to fix the constructor, and _then_ adding my other changes. That should give us the best of both worlds :D